### PR TITLE
feat(core,btp): updated button with new features from styles

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Related Issue(s)

closes

## Description

- Added BTP Button documentation section, which includes:
  - new `toolHeader` type
  - new `nested` type
  - new `nested-square` type
- Added badge ability to the ordinary Core button
- Added `vertical` mode to the segmented button

BREAKING CHANGE: The tool header button now comes from the `@fundamental-ngx/btp/button`

### CODE REVIEW

**Review by mikerodonnell89 — COMMENTED** [review]
Correct me if I'm wrong but I would think different naming (i.e. the new ButtonHostDirective could be called ButtonComponent, and ButtonComponent would get a new name) would allow us to preserve the existing core button component api and not have to introduce breaking changes

**Comment by mikerodonnell89** [comment]
Replying to myself: I suppose now is the time for breaking changes to insure best practices in the future

File: libs/core/src/lib/button/button-host.directive.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    /** @hidden */
+    specialButtonType: Array<string> = ['emphasized', 'positive', 'negative', 'attention'];
+
+    /** @hidden */
+    elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
+
+    /** @hidden */
+    class = signal('');
```
Comment (N1XUS) [inline libs/core/src/lib/button/button-host.directive.ts:116]: Why not keep the input properties like `class`, `type`, `fdType` as is, without the `_` prefix, and the signals will have `_` prefix instead?

File: libs/core/src/lib/button/button-host.directive.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    );
+
+    /** @hidden */
+    _classNamePredicates = signal<Array<() => string | string[] | null>>([
+        () => 'fd-button',
+        this.class,
+        computed(() => (this.disabled() || this.ariaDisabled() ? 'is-disabled' : '')),
+        computed(() => (this.fdType() ? `fd-button--${this.fdType()}` : '')),
+        computed(() => (this.fdMenu() ? 'fd-button--menu' : '')),
+        computed(() => (this.toggled() ? `fd-button--toggled` : ''))
```
Comment (N1XUS) [inline libs/core/src/lib/button/button-host.directive.ts:155]:
```typescript
computed(() => {
    return [
        'fd-button', 
        this.class(),
        this.disabled() || this.ariaDisabled() ? 'is-disabled' : '',
        this.fdType() ? `fd-button--${this.fdType()}` : '',
        this.fdMenu() ? 'fd-button--menu' : '',
        this.toggled() ? `fd-button--toggled` : ''
    ]
});
```

**Review by N1XUS — APPROVED** [review]

File: libs/btp/button/src/button.component.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    @Input()
+    endGlyph: Nullable<string> = '';
+
+    /** Whether the button is the tool header button */
+    @Input({ transform: coerceBooleanProperty })
+    toolHeader: BooleanInput;
+
+    /** Whether the button is nested in the tool header */
+    @Input({ transform: coerceBooleanProperty })
+    nested: BooleanInput;
```
[the remark this answers is not in the document]
Comment (InnaAtanasova) [reply libs/btp/button/src/button.component.ts:57]: `fd-button` has `fdType` to specify if the button is transparent, positive, ghost, etc.
nested, toolHeader are also types of the button. I think we can add them to the `fdType` or add a new `fdbType`. For the "expander" type we can have "expander" and "nested-expander" or something like this.

File: libs/btp/navigation/navigation-list/navigation-list-item.component.html [hunk 4]
```diff
@@ -20,8 +20,9 @@
     <button
         (click)="toggle()"
         *ngIf="childNavigationListComponent() && normalizedLevel() < 3 && linkComponent?._routerLink"
-        fd-button
-        class="fd-button--nested fd-navigation__expander"
+        fdb-button
+        nested
+        class="fd-navigation__expander"
```
[the remark this answers is not in the document]
Comment (g-cheishvili) [reply libs/btp/navigation/navigation-list/navigation-list-item.component.html:25]: it is navigation's button, not the button one, so if we want to move it into the input, it should not be moved into the button and definitely should be done in different PR

File: libs/btp/button/src/button.component.scss [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        right: -0.125rem;
+        top: -0.125rem;
+        position: absolute;
+        border-radius: 100%;
+        background: var(--sapContent_BadgeBackground);
+        color: var(--sapContent_BadgeBackground);
+        border: 0.0625rem solid var(--fdButton_ToolHeader_Badge_Border_Color);
+        overflow: hidden;
+    }
+}
```
[the remark this answers is not in the document]
Comment (g-cheishvili) [reply libs/btp/button/src/button.component.scss:27]: https://github.com/SAP/fundamental-styles/pull/4925
This needs to be merged first then

**Comment by g-cheishvili** [comment]
Waiting for styles PR, this will be better

**Comment by InnaAtanasova** [comment]
Screenshot 2023-10-31 at 7 21 47 PM

why are these buttons in BTP? They are Core buttons. Only Tool Header button and Nested Button are BTP specific

**Comment by g-cheishvili** [comment]
No longer contains BREAKING CHANGE

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**